aspnetcore/host-and-deploy/aspnet-core-module.md

```diff
@@ -660,6 +415,617 @@ The Hosting Bundle installer logs for the module are found at *C:\\Users\\%UserN
 
+::: moniker-end
+
+::: moniker range="= aspnetcore-2.2"
+
+The ASP.NET Core Module is a native IIS module that plugs into the IIS pipeline to either:
+
+* Host an ASP.NET Core app inside of the IIS worker process (`w3wp.exe`), called the [in-process hosting model](#in-process-hosting-model).
+* Forward web requests to a backend ASP.NET Core app running the [Kestrel server](xref:fundamentals/servers/kestrel), called the [out-of-process hosting model](#out-of-process-hosting-model).
+
+Supported Windows versions:
+
+* Windows 7 or later
+* Windows Server 2008 R2 or later
+
+When hosting in-process, the module uses an in-process server implementation for IIS, called IIS HTTP Server (`IISHttpServer`).
+
+When hosting out-of-process, the module only works with Kestrel. The module is incompatible with [HTTP.sys](xref:fundamentals/servers/httpsys).
+
+## Hosting models
+
+### In-process hosting model
+
+To configure an app for in-process hosting, add the `<AspNetCoreHostingModel>` property to the app's project file with a value of `InProcess` (out-of-process hosting is set with `OutOfProcess`):
+
+```xml
+<PropertyGroup>
+  <AspNetCoreHostingModel>InProcess</AspNetCoreHostingModel>
+</PropertyGroup>
+```
+
+The in-process hosting model isn't supported for ASP.NET Core apps that target the .NET Framework.
+
+If the `<AspNetCoreHostingModel>` property isn't present in the file, the default value is `OutOfProcess`.
+
+The following characteristics apply when hosting in-process:
+
+* IIS HTTP Server (`IISHttpServer`) is used instead of [Kestrel](xref:fundamentals/servers/kestrel) server. For in-process, [CreateDefaultBuilder](xref:fundamentals/host/web-host#set-up-a-host) calls <xref:Microsoft.AspNetCore.Hosting.WebHostBuilderIISExtensions.UseIIS*> to:
+
+  * Register the `IISHttpServer`.
+  * Configure the port and base path the server should listen on when running behind the ASP.NET Core Module.
+  * Configure the host to capture startup errors.
+
+* The [requestTimeout attribute](#attributes-of-the-aspnetcore-element) doesn't apply to in-process hosting.
+
+* Sharing an app pool among apps isn't supported. Use one app pool per app.
+
+* When using [Web Deploy](/iis/publish/using-web-deploy/introduction-to-web-deploy) or manually placing an [app_offline.htm file in the deployment](xref:host-and-deploy/iis/index#locked-deployment-files), the app might not shut down immediately if there's an open connection. For example, a websocket connection may delay app shut down.
+
+* The architecture (bitness) of the app and installed runtime (x64 or x86) must match the architecture of the app pool.
+
+* If setting up the app's host manually with `WebHostBuilder` (not using [CreateDefaultBuilder](xref:fundamentals/host/web-host#set-up-a-host)) and the app is ever run directly on the Kestrel server (self-hosted), call `UseKestrel` before calling `UseIISIntegration`. If the order is reversed, the host fails to start.
```

Comment:
I'm pretty sure the comment about how reversing the order of `UseKestrel` and `UseIISIntegration` breaks things hasn't been true for a while. You fixed that, right @Tratcher?

Reply:
Also, does this really belong in the "In-process hosting model" section? I guess this is about self-hosting apps that are normally hosted in-proc by ancm, so it makes a little bit a sense, but it's also a little confusing.

Reply:
Yeah, we dealt with that a long time ago. Maybe it was this one? https://github.com/aspnet/IISIntegration/issues/242